### PR TITLE
Proxy creation

### PR DESCRIPTION
**Pull request by Heavy32**

### CODE REVIEW

File: Proxy/InternetAdressRequester.cs [hunk 1]
```diff
@@ -0,0 +1,21 @@
+﻿using System;
+
+namespace Decorator
+{
+    class InternetAdressRequester : IRequester
```
Comment (alexzaitzev) [inline Proxy/InternetAdressRequester.cs]: You shouldn't create your own class. Just use base class [HttpWebRequest](https://docs.microsoft.com/en-us/dotnet/api/system.net.httpwebrequest?view=netcore-3.1)

File: Proxy/Proxy.csproj [hunk 2]
```diff
@@ -0,0 +1,7 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
```
Comment (alexzaitzev) [inline Proxy/Proxy.csproj]: dotnet3.1

**Review by alexzaitzev — CHANGES_REQUESTED** [review]
Address the comments please

File: Proxy/InternetAdressRequester.cs (hunk 1 above)
Comment (Heavy32) [reply Proxy/InternetAdressRequester.cs]: done

File: Proxy/Proxy.csproj (hunk 2 above)
Comment (Heavy32) [reply Proxy/Proxy.csproj]: done

File: Decorator/Program.cs [hunk 3]
```diff
@@ -1,5 +1,4 @@
 ﻿using System;
-using System.Text;
 
 namespace Decorator
```
Comment (alexzaitzev) [inline Decorator/Program.cs]: Redundant file?

File: Patterns.sln [hunk 4]
```diff
@@ -11,7 +11,9 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Adapter", "Adapter\Adapter.
 EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Singleton", "Singleton\Singleton.csproj", "{F31C4C95-1310-4521-A1E4-0506915DE33B}"
 EndProject
-Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Factory", "Factory\Factory.csproj", "{6604026A-7C11-4D1D-84C2-3E030F674C1B}"
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Factory", "Factory\Factory.csproj", "{6604026A-7C11-4D1D-84C2-3E030F674C1B}"
```
Comment (alexzaitzev) [inline Patterns.sln]: Please , go through your PRs carefully and delete unnecessary files/changes at least

File: Proxy/Proxy.cs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Proxy
+{
+    class Proxy : WebRequest
+    {
+        private readonly Request request;
+
+        public Proxy()
+        {
+            request = new Request();
```
Comment (alexzaitzev) [inline Proxy/Proxy.cs]: If you create new object in ctor - it's not a proxy

File: Proxy/Proxy.csproj [hunk 6]
```diff
@@ -0,0 +1,10 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netcoreapp3.1</TargetFramework>
+    <ApplicationIcon />
```
Comment (alexzaitzev) [inline Proxy/Proxy.csproj]: Why this line is here?

File: Proxy/Proxy.csproj [hunk 7]
```diff
@@ -0,0 +1,10 @@
...
+    <OutputType>Exe</OutputType>
+    <StartupObject />
```
Comment (alexzaitzev) [inline Proxy/Proxy.csproj]: For what purpose it's here?

File: Proxy/Request.cs [hunk 8]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Net;
+
+namespace Proxy
+{
+    public class Request : WebRequest
```
Comment (alexzaitzev) [inline Proxy/Request.cs]: Why do we need `Request` class?

**Review by alexzaitzev — CHANGES_REQUESTED** [review]

File: Proxy/Request.cs (hunk 8 above)
Comment (Heavy32) [reply Proxy/Request.cs]: don't need

File: Proxy/Proxy.csproj (hunk 7 above)
Comment (Heavy32) [reply Proxy/Proxy.csproj]: done

File: Proxy/Proxy.csproj (hunk 6 above)
Comment (Heavy32) [reply Proxy/Proxy.csproj]: I have no clue mate...

File: Proxy/Proxy.cs (hunk 5 above)
Comment (Heavy32) [reply Proxy/Proxy.cs]: done

File: Decorator/Program.cs (hunk 3 above)
Comment (Heavy32) [reply Decorator/Program.cs]: done

File: Patterns.sln (hunk 4 above)
Comment (Heavy32) [reply Patterns.sln]: done

File: Proxy/Proxy.cs [hunk 9]
```diff
@@ -0,0 +1,21 @@
+﻿using System.Net;
+
+namespace Proxy
+{
+    class Proxy : WebRequest
```
Comment (alexzaitzev) [inline Proxy/Proxy.cs]: WebRequestProxy?

Comment (Heavy32) [reply Proxy/Proxy.cs]: Yeah, sure

**Pull request merged**